script/cache_critical_dns

```diff
@@ -17,9 +15,153 @@ CRITICAL_HOST_ENV_VARS = %w{
   DISCOURSE_REDIS_SLAVE_HOST
   DISCOURSE_REDIS_REPLICA_HOST
 }
+HOST_RESOLVER_CACHE = {}
+HOST_HEALTHY_CACHE = {}
+HOSTS_PATH = ENV['DISCOURSE_HOSTS_FILE'] || "/etc/hosts"
```

Comment:
Could we use a common prefix for these override envs?  `DISCOURSE_DNS_CACHE_`, which you used below, seems perfectly reasonable.